src/main/java/uk/gov/cshr/service/security/IdentityService.java

```diff
@@ -168,13 +168,34 @@ public Identity setLastLoggedIn(Instant datetime, Identity identity) {
         return identityRepository.save(identity);
     }
 
-    public void updateEmailAddressAndEmailRecentlyUpdatedFlagToTrue(Identity identity, String email) {
+    public void updateEmailAddressWhereNewEmailIsAgency(Identity identity, String email, AgencyToken newAgencyToken) {
         Identity savedIdentity = identityRepository.findById(identity.getId())
                 .orElseThrow(() -> new IdentityNotFoundException("No such identity: " + identity.getId()));
 
+        if (newAgencyToken.getUid() != null) {
+            log.debug("Updating agency token for user: oldAgencyToken = {}, newAgencyToken = {}", identity.getAgencyTokenUid(), newAgencyToken.getUid());
+            savedIdentity.setAgencyTokenUid(newAgencyToken.getUid());
+        }
+
         savedIdentity.setEmail(email);
         savedIdentity.setEmailRecentlyUpdated(true);
```

Comment:
do you need to update spring with this?

Reply:
We're updating spring in the emailUpdateService.java method which calls this:

`public void updateEmailAddress(HttpServletRequest request, Identity identity, EmailUpdate emailUpdate)`

Is this what you had in mind?

Reply:
It was to do with the boolean issue, spring lazy loads the boolean value, i.e. it doesn't get the updated value straight away.  This is why I was asking.